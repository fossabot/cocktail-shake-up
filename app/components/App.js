/* @flow */

import React, { PureComponent } from 'react';
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom';
import { Helmet } from 'react-helmet';
import type { Node } from 'react';
import { connect } from 'react-redux';
import './App.css';

export type Props = {};
export type State = {};

export class App extends PureComponent<Props, State> {
  render(): Node {
    return (
      <Router>
        <div>
          <Switch>
            <Route
              path="/goodbye"
              render={() => (
                <div id="goodbye">
                  <p>Goodbye World</p>
                  <Helmet>
                    <title>Goodbye Cocktail Shake Up</title>
                  </Helmet>
                </div>
              )}
            />
            {/* Most generic path always goes last */}
            <Route
              path="/"
              render={() => (
                <div id="greet">
                  <p>Hello World</p>
                  <Helmet>
                    <title>Hello Cocktail Shake Up</title>
                  </Helmet>
                </div>
              )}
            />
          </Switch>
        </div>
      </Router>
    );
  }
}

const mapStateToProps = state => {
  return {
    notification: state.notification,
    locale: state.locale,
  };
};

export default connect(mapStateToProps)(App);
