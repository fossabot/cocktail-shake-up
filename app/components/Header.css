header {
  display: flex;

  background-color: skyblue;
}

header * {
  margin: 0;

  text-align: center;

  color: white;

  line-height: 3rem;

  flex: 1;
}

header *:first-child {
  margin: 0 0 0 1rem;

  text-align: left;
  text-transform: capitalize;
}

header *:last-child {
  margin: 0 1rem 0 0;

  text-align: right;
  text-transform: capitalize;
}

header .path {
  text-transform: capitalize;

  color: darkred;

  font-weight: 600;
}
