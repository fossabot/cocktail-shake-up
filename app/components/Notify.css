.notification {
  position: fixed;
  bottom: 0;

  width: 100%;
  height: 0;

  transition: height 0.25s, opacity 0.5s;
  text-align: center;

  opacity: 0;
  color: white;
  background-color: grey;
}

.notification.visible {
  height: auto;
  min-height: 8rem;
  padding: 1rem 0 1rem 0;

  transition: height 0.25s, opacity 0.5s;

  opacity: 1;
}

.notification .close {
  float: right;

  width: 2.4rem;
  height: 2rem;
  margin: 0 1rem;

  color: darkred;
}

.notification h1 {
  color: transparent;
}

.notification.visible a {
  color: white;
}

.notification.visible h1 {
  margin: 0 0 1rem;

  color: white;
}
