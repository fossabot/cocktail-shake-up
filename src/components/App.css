:root {
  font-size: 14px;

  --ui-big-button-width: 8rem;
  --ui-table-image-size: 6rem;
}

body {
  width: 100vw;
  height: 100vh;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Helvetica Neue", sans-serif;
}

.big-button {
  width: var(--ui-big-button-width);
  height: 2.5rem;
  font-size: initial;
  font-weight: 600;
  line-height: 2rem;
  text-transform: uppercase;
  color: white;
  border: 0;
  background-color: darkred;

  fill: currentColor;
}

.big-button:hover {
  cursor: pointer;
}

.not-found {
  text-align: center;
}

.not-found::after {
  display: inline-block;
  width: 2.4rem;
  height: 2rem;
  content: "";
  background-size: cover;
}


@media (max-width: 376px) {
  :root {
    font-size: 12px;
  }

  img {
    display: none;
  }
}

@media (min-width: 768px) {
  :root {
    font-size: 15px;
  }
}

@media (min-width: 1200px) {
  :root {
    font-size: 18px;
  }
}
