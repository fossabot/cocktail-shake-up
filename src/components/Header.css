header {
  display: flex;
  background-color: skyblue;
}

header * {
  flex: 1;
  margin: 0;
  line-height: 3rem;
  text-align: center;
  color: white;
}

header *:first-child {
  margin: 0 0 0 1rem;
  text-align: left;
  text-transform: capitalize;
}

header *:last-child {
  margin: 0 1rem 0 0;
  text-align: right;
  text-transform: capitalize;
}

header .path {
  font-weight: 600;
  text-transform: capitalize;
  color: darkred;
}
