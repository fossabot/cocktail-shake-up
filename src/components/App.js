/* @flow */

import React, { PureComponent } from 'react';
import { func, object } from 'prop-types';
import { connect } from 'react-redux';
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom';
import './App.css';

export class App extends PureComponent {
  static propTypes = {
    dispatch:     func,
    locale:       object,
    notification: object,
  };

  render() {
    return (
      <Router>
        <div>
          <Switch>
            <Route path="/" render={() => <div>Hello World</div>} />
          </Switch>
        </div>
      </Router>
    );
  }
}

const mapStateToProps = state => {
  return {
    notification: state.notification,
    locale:       state.locale,
  };
};

export default connect(mapStateToProps)(App);
