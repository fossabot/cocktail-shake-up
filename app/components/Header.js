/* @flow */

import { Link } from 'react-router-dom';
import React from 'react';
import './Header.css';

const Header = (props: Object) => {
  return (
    <header>
      <Link to={`/${props.previous}`}>{props.previousTitle ? props.previousTitle : props.previous}</Link>
      <p className="path">{props.path}</p>
      <Link to={`/${props.next}`}>{props.nextTitle ? props.nextTitle : props.next}</Link>
    </header>
  );
};

export default Header;
