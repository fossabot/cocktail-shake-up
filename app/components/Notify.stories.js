import Notify from './Notify';
import React from 'react';
import { storiesOf } from '@storybook/react';

storiesOf('Header', module)
  .add('Default', () => <Notify />)
  .add('When Visible', () => <Notify visible={true} />)
  .add('With Close Handler', () => <Notify onCloseHandler={() => {}} visible={true} />)
  .add('With Title', () => <Notify visible={true} title="A Title" />)
  .add('With Title and Body', () => <Notify visible={true} body="A Body" title="A Title" />)
  .add('With Title, Body and Close Handler', () => (
    <Notify onCloseHandler={() => {}} visible={true} body="A Body" title="A Title" />
  ));
