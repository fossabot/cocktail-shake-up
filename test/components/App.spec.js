/* @flow */
import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../../src/components/App';

test('should render an App', () => {
  const tree = shallow(<App />).debug();
  expect(tree).toMatchSnapshot();
});
