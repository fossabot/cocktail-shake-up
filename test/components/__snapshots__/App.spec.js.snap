// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an App 1`] = `
"<BrowserRouter>
<div>
<Switch>
<Route path=\\"/\\" render={[Function]} />
</Switch>
</div>
</BrowserRouter>"
`;
